src/dbus.c

```diff
@@ -0,0 +1,89 @@
+#include "dbus.h"
+#include "refresh_status.h"
+
+static void
+handle_notifications_method_call(GDBusConnection       *connection,
+                                 const gchar           *sender,
+                                 const gchar           *object_path,
+                                 const gchar           *interface_name,
+                                 const gchar           *method_name,
+                                 GVariant              *parameters,
+                                 GDBusMethodInvocation *invocation,
+                                 gpointer               user_data)
+{
+    if (g_strcmp0 (interface_name, "io.snapcraft.SnapDesktopIntegration") != 0) {
+        g_dbus_method_invocation_return_error(invocation, G_DBUS_ERROR, G_DBUS_ERROR_UNKNOWN_INTERFACE, "Unknown interface");
+        return;
+    }
+    if (g_strcmp0 (method_name, "ApplicationIsBeingRefreshed") == 0) {
+        handle_application_is_being_refreshed(parameters, (DsState*) user_data);
+        g_dbus_method_invocation_return_value(invocation, NULL);
+        return;
+    }
+    g_dbus_method_invocation_return_error(invocation, G_DBUS_ERROR, G_DBUS_ERROR_UNKNOWN_METHOD, "Unknown method");
+}
+
+bool
+register_dbus (GDBusConnection  *connection,
+               DsState          *state,
+               GError          **error)
+{
+    static const gchar introspection_xml[] =
+    "<node>"
+    "    <!-- io.snapcraft.SnapDesktopIntegration:"
+    "        @short_description: Interface for desktop integration of Snapd"
+    ""
+    "        This D-Bus interface allows to achieve better integration of"
```

Comment:
Unless I'm terribly mistaken, these will be concatenated as "...integration of SnapdThis D-Bus interface..." unless you explicitly write a `\n` at the end of each line.

Reply:
Ops... you are right!

Reply:
Anyway, being a description, I don't think it is problematic: it is there just as "documentation", but the XML is directly consumed by GLib.